metrics-core/src/lib.rs

```diff
@@ -59,3 +60,28 @@ pub trait MetricsRecorder {
     /// of the underlying observed values, and callers will need to process them accordingly.
     fn record_histogram<K: AsRef<str>>(&mut self, key: K, values: &[u64]);
 }
+
+/// A value that holds a point-in-time view of collected metrics.
+pub trait MetricsSnapshot {
```

Comment:
prob can drop the word metric from here?

Reply:
Yeah, it's probably reasonable to drop `Metrics` from both the snapshot trait _and_ recorder trait.  I'll do that.